runtime/src/chpl-topo.c

```diff
@@ -310,6 +310,58 @@ void chpl_topo_setMemSubchunkLocality(void* p, size_t size,
 }
 
 
+void chpl_topo_touchMemFromSubloc(void* p, size_t size, chpl_bool onlyInside,
+                                  c_sublocid_t subloc) {
+  size_t pgSize;
+  unsigned char* pPgLo;
+  size_t nPages;
+  hwloc_cpuset_t cpuset;
+  int flags;
+
+  _DBG_P("chpl_topo_touchMemFromSubloc(%p, %#zx, onlyIn=%s, %d)\n",
+         p, size, (onlyInside ? "T" : "F"), (int) subloc);
+
+  if (!haveTopology
+      || !topoSupport->cpubind->get_thread_cpubind
+      || !topoSupport->cpubind->set_thread_cpubind) {
+    return;
+  }
+
+  alignAddrSize(p, size, onlyInside, &pgSize, &pPgLo, &nPages);
+
+  _DBG_P("    localize %p, %#zx bytes (%#zx pages)\n",
+         pPgLo, nPages * pgSize, nPages);
+
+  if (nPages == 0)
+    return;
+
+  if ((cpuset = hwloc_bitmap_alloc()) == NULL) {
+    report_error("hwloc_bitmap_alloc()", errno);
+  }
+
+  flags = HWLOC_CPUBIND_THREAD;
+  if (hwloc_get_thread_cpubind(topology, pthread_self(), cpuset, flags)) {
```

Comment:
The HWLOC_CPUBIND_THREAD flag is not needed here because the thread-specific function hwloc_get_thread_cpubind() is being used.  However, I would suggest keeping the flag and switching to using hwloc_get_cpubind() instead.  That would keep the abstraction level slightly higher by not having to refer to pthreads here.

But, there is another case like this in chpl_topo_getThreadLocality() which I missed earlier (sorry), so if you'd rather not make the change here and keep the two consistent, that's OK.